Prepare release 3.1.1 , retail-react-app: 2.0.0, commerce-sdk-react: 1.0.2

shethj opened the pull request:
<!--- Provide a short summary of your changes in the Title field above -->
This release includes the following changes:

- V3 Fix checkout card number #1424
- commerce-sdk-react: have typedoc-related deps as dev dependencies #1425
- Fix: resolve performance issue due to webpack stats #1391
- commerce-sdk-react: Update commerce-sdk-isomorphic to v1.10.4
- Updated lockfileVersion to 3 for Node 18 and npm 9

And

- Releases monorepo v3.1.1
- Releases retail-react-app: v2.0.0
- commerce-sdk-react: v1.0.2

# Description

<!--- A longer summary of your changes, including: a description of the issue that you’re addressing, a list of required dependencies (if applicable), and any other relevant context. -->

# Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] **Bug fix** (non-breaking change that fixes an issue)
- [ ] **New feature** (non-breaking change that adds functionality)
- [ ] **Documentation update**
- [ ] **Breaking change** (could cause existing functionality to not work as expected)
- [ ] **Other changes** (non-breaking changes that does not fit any of the above)

> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- (step1)

# Checklists

<!--- Enter an `x` in all the boxes that apply. -->
<!--- If you’re unsure about any of these, don’t hesitate to ask. We’re here to help! -->

## General

- [ ] Changes are covered by test cases
- [ ] CHANGELOG.md updated with a short description of changes (_not_ required for documentation updates)

## Accessibility Compliance

You must check off all items in **one** of the follow two lists:

- [ ] There are no changes to UI

_or..._

- [ ] Changes were tested with a Screen Reader (iOS VoiceOver or Android Talkback) and had no issues
- [ ] Changes comply with [WCAG 2.0 guidelines levels A and AA](https://www.wuhcag.com/wcag-checklist/)
- [ ] Changes to common UI patterns and interactions comply with [WAI-ARIA best practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## Localization

- [ ] Changes include a UI text update in the Retail React App (which requires translation)

adamraya committed:
Use maskedNumber

vmarta committed:
commerce-sdk-react: have typedoc-related deps as dev

shethj committed:
Bugfix: resolve performance issue due to webpack stats

vmarta commented on packages/commerce-sdk-react/package.json:
```diff
@@ -42,12 +42,10 @@
-    "jwt-decode": "^3.1.2",
-    "typedoc": "^0.24.7",
-    "typedoc-plugin-missing-exports": "^2.0.0"
+    "jwt-decode": "^3.1.2"
```
Ok, so we do need to release commerce-sdk-react too, because of fixing the typedoc dependencies 👍

vmarta commented on packages/commerce-sdk-react/CHANGELOG.md:
```diff
@@ -1,3 +1,7 @@
+## v1.0.2-preview.1 (Sep 13, 2023)
+## v1.0.2-preview.0 (Sep 13, 2023)
+## v3.1.1-preview.0 (Sep 13, 2023)
+## v1.0.2-preview.0 (Sep 13, 2023)
```
Let's mention and add a line about fixing the typedoc dependencies.

And please also clean up the headings before merging. There's a known issue with the bump-version script that adds multiple headings like this.

vmarta commented on packages/template-retail-react-app/package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@salesforce/retail-react-app",
-  "version": "1.0.1",
+  "version": "2.0.0-preview.1",
```
For retail-react-app, let's update the changelog, as Adam had done it in his PR: https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1424/files#diff-3aec2f678f9138a6279a6c8b3ac8c3617136aef761dc310044a4e75487e75996

vmarta commented on packages/pwa-kit-dev/CHANGELOG.md:
```diff
@@ -1,3 +1,4 @@
+## v3.1.1-preview.0 (Sep 13, 2023)
```
Let's update the changelog here to mention about Joel's fix.

vmarta commented on packages/commerce-sdk-react/package-lock.json:
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@salesforce/commerce-sdk-react",
   "version": "1.0.1",
```
This package version should be `1.0.2-preview.1` → that's probably why CI fails when `npm ci` this package and complains that it cannot find the lock file. Because the `version` is mismatched between package and package lock.

shethj committed:
Revert version bump

shethj committed:
Update code to consume commerce-sdk-isomorphic

shethj committed:
Bump retail app version

shethj committed:
bump pwa-kit version to 3.1.1-preview.0

shethj committed:
Update all lockfileVersion to 3

shethj committed:
Bump version for release

shethj committed:
Updated changelog

shethj committed:
Updated README to recommend node 18

shethj committed:
Ignore typecheck errors in test-commerce-sdk-react

vmarta commented on packages/pwa-kit-react-sdk/CHANGELOG.md:
```diff
@@ -1,3 +1,5 @@
+## v3.1.1 (Sep 21, 2023)
+## v3.1.1-preview.0 (Sep 21, 2023)
```
Please make sure all changelog files do not have lines mentioning the temporary preview versions.

shethj committed:
Remove preview versions from changelog

alexvuong commented on packages/commerce-sdk-react/src/hooks/ShopperLogin/query.ts:
```diff
@@ -13,42 +13,6 @@ import * as queryKeyHelpers from './queryKeyHelpers'
- * @see {@link https://salesforcecommercecloud.github.io/commerce-sdk-isomorphic/classes/shopperlogin.shopperlogin-1.html#retrievecredqualityuserinfo | `commerce-sdk-isomorphic` documentation} for more information on the parameters and returned data type.
- * @see {@link https://tanstack.com/query/latest/docs/react/reference/useQuery | TanStack Query `useQuery` reference} for more information about the return value.
- */
-export const useCredQualityUserInfo = (
```
Are we not using this ?

shethj replied:
We were not using it anywhere and the endpoint itself has been removed from SCAPI.
The API changes are released as well.

bfeister reviewed (changes requested):
Blocking pre-emptively only because I want to check on a few things before we 🚀

bfeister reviewed (changes requested):
The following changeset (which can go to retail react app CHANGELOG) should be cherry-picked as they don't pertain to storefront preview

```
- Cleanup wrong import path for page-designer component [#1441](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1441) @clavery
- Modularize country code source for targeting via extensibility [#1445](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1445) @bfeister
- Export icon helper function for target via overrides [#1420](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1420) @cbrother-csu
- Migrate Page Designer core types to commerce-sdk-react [#1441](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1441) @clavery
```

bfeister committed:
Cherry pick 4 commits from `develop`

- Cleanup wrong import path for page-designer component [#1441](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1441) @clavery
- Modularize country code source for targeting via extensibility [#1445](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1445) @bfeister
- Export icon helper function for target via overrides [#1420](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1420) @cbrother-csu
- Migrate Page Designer core types to commerce-sdk-react [#1441](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1441) @clavery

bfeister reviewed (approved):
Cherry picked the following. We're good now 🚀 

```
- Cleanup wrong import path for page-designer component [#1441](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1441) @clavery
- Modularize country code source for targeting via extensibility [#1445](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1445) @bfeister
- Export icon helper function for target via overrides [#1420](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1420) @cbrother-csu
- Migrate Page Designer core types to commerce-sdk-react [#1441](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1441) @clavery
```
